commit 5ffa0dbfdc9fc05acae02d5b0dc766ec778569ac
Author: Dawid Niedzwiecki <dawidn@google.com>
Date:   Fri Dec 6 09:15:13 2024 +0000

    platform/chrome: cros_ec: jump to RW before probing
    
    There are EC devices, like FPMCU, that use RWSIG as a method of
    authenticating RW section. After the authentication succeeds, EC device
    waits some time before jumping to RW. EC can be probed before the jump,
    which means there is a time window after jump to RW in which EC won't
    respond, because it is not initialized. It can cause a communication
    errors after probing.
    
    To avoid such problems, send the RWSIG continue command first, which
    skips waiting for the jump to RW. Send the command more times, to make
    sure EC is ready in RW before the start of the actual probing process. If
    a EC device doesn't support the RWSIG, it will respond with invalid
    command error code and probing will continue as usual.
    
    Signed-off-by: Dawid Niedzwiecki <dawidn@google.com>
    Link: https://lore.kernel.org/r/20241206091514.2538350-2-dawidn@google.com
    Signed-off-by: Tzung-Bi Shih <tzungbi@kernel.org>

diff --git a/drivers/platform/chrome/cros_ec_spi.c b/drivers/platform/chrome/cros_ec_spi.c
index 86a3d32a7763..946373238502 100644
--- a/drivers/platform/chrome/cros_ec_spi.c
+++ b/drivers/platform/chrome/cros_ec_spi.c
@@ -766,7 +766,7 @@ static int cros_ec_spi_probe(struct spi_device *spi)
 	ec_dev->din_size = EC_MSG_PREAMBLE_COUNT +
 			   sizeof(struct ec_host_response) +
 			   sizeof(struct ec_response_get_protocol_info);
-	ec_dev->dout_size = sizeof(struct ec_host_request);
+	ec_dev->dout_size = sizeof(struct ec_host_request) + sizeof(struct ec_params_rwsig_action);
 
 	ec_spi->last_transfer_ns = ktime_get_ns();